src/phoenix/trace/instrumentor.py

```diff
@@ -0,0 +1,24 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+
+@dataclass(frozen=True)
+class Instrumentor(ABC):
```

Comment:
Would duck typing work? Just asking because `_raise_import_error` seems like a helper function that doesn't need to be in an `ABC`, and if we remove it, duck typing can work pretty well here. I see `otel` uses inheritance, but we don't really need follow it.
```suggestion
class Instrumentor(Protocol):
```

Reply:
Yeah I was just blindly following the Model base class but agree it's probably a relatively useless abstraction. Nixed.